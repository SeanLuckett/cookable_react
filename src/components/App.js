import React, {Component} from 'react';

import Grid from '@material-ui/core/Grid';
import AppBar from '@material-ui/core/AppBar';
import ToolBar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

import {createMuiTheme, MuiThemeProvider} from '@material-ui/core/styles';
import RecipeListContainer from "./RecipeListContainer";


const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#1a1a59',
    },
    secondary: {
      main: '#ffdbaa'
    }
  }
});

class App extends Component {
  render() {
    return (
      <MuiThemeProvider theme={theme}>
        <Grid container justify-xs-center>
          <Grid item sm={12}>
            <AppBar>
              <ToolBar>
                <Typography variant='title'
                            color='secondary'>Cookable</Typography>
              </ToolBar>
            </AppBar>
          </Grid>

          <Grid item sm={12}>
            <RecipeListContainer/>
          </Grid>
        </Grid>
      </MuiThemeProvider>
    );
  }
}

export default App;
