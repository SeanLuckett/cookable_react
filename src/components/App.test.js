import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  const app = shallow(<App/>);

  it('renders', () => {
    expect(app).toMatchSnapshot();
  });

  it('contains a connected RecipeListContainer', () => {
    expect(app.find('Connect(RecipeListContainer)').exists()).toBe(true);
  });
});