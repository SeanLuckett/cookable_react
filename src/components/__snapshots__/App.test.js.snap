// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div
  className="page-wrap"
>
  <section
    className="titlebar"
  >
    <h1
      className="header"
    >
      Cookable
    </h1>
  </section>
  <section
    className="main-content"
  >
    <h1
      className="section-title"
    >
      Recipes
    </h1>
    <table
      className="recipe-list"
    >
      <thead>
        <tr>
          <th>
            Title
          </th>
          <th>
            Type
          </th>
          <th>
            Prep time
          </th>
          <th>
            Source
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Some Title
          </td>
          <td>
            Entree
          </td>
          <td>
            Under 30 min
          </td>
          <td>
            Epicurious
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</div>
`;
